uberfire-api/src/main/java/org/uberfire/workbench/events/NotificationEvent.java

```diff
@@ -29,6 +29,9 @@
     private boolean isSingleton;
     private PlaceRequest placeRequest;
     private Integer initialTopOffset;
+    private boolean autoHide = true;
```

Comment:
I think it would be better to set these new attributes in a constructor instead of using `set`methods.

Reply:
I don't think it's a good idea.
There are already 6 constructors, and if I add a new one I have to set all default parameters of previous constructors. Default values should be set by class initialization, not by external classes, and if default values change in future, we have to change in all objects creation too.
Also, a constructor with 8 parameters (current 5 + new 3) is not a good idea.

Example:
```Java
// if I want to set only A and E, I must set also B, C and D
// even though I don't know what is the default value of these parameters
public class X {
    X(String A, String B, String C, String D, String E) { }
}

// when parameters have different types we can create different constructors
// with same length of parameters
// This is not possible when parameters have same type
public class X {
    X(int A) { }
    X(String B) { }
    X(int A, String B) { } // this is fine
    X(int A, String C) { } // this is not possible, don't know which constructor to call
}

// Problems:
//     don't want to set all parameters
//     it's not my responsibility
//     if default values changes in class impl, we have to change here too
//     constructor is too big
X x = new X("A", ?, ?, ?, "E");

// this is much better or something else like Builder Pattern
X x = new X().setA("A").setB("B");
```

Reply:
@gabriel376 Ok, I agree with you. Maybe when we have named parameters and default values in Java we can change that. :)